src/lib/viz/source/SQLSource.ts

```diff
@@ -226,11 +227,14 @@ export class SQLSource extends Source {
 
   private extractMetadataFrom(mapInstance: MapInstance) {
     const { stats } = mapInstance.metadata.layers[0].meta;
-    const geometryType = parseGeometryType(stats.geometryType);
-    const fieldStats = this.getCompleteFieldStats(stats);
+    let geometryType: GeometryType | undefined;
 
-    const metadata = { geometryType, stats: fieldStats };
+    if (stats.geometryType) {
```

Comment:
Maybe it easier to do something like:

```
if (!stats.geometryType) {
  return { 'Point', stats: [] }
}
```

And even better, if you encapsulate the check `!stats.geometryType` in a method called `hasData`.

I would expect you don't need to check if `geometryType` exists and it is not going to fail for that reason in other places

Reply:
Same thoughts here around 'hasData' or 'isEmpty' method